lib/openhab/dsl/items/image_item.rb

```diff
@@ -0,0 +1,157 @@
+        def_item_delegator :@image_item
+
+        item_type Java::OrgOpenhabCoreLibraryItems::ImageItem
+
+        #
+        # Creates a new ImageItem
+        #
+        # @param [Java::OrgOpenhabCoreLibraryItems::ImageItem] image_item
+        #   The OpenHAB ImageItem to delegate to
+        #
+        def initialize(image_item)
+          logger.trace("Wrapping #{image_item}")
+          @image_item = image_item
+
+          item_missing_delegate { @image_item }
+
+          super()
+        end
+
+        #
+        # Update the image
+        #
+        # @param [String] base_64 base_64 encoding of an image,
+        #   if not supplied keyword_args are used to find update
+        # @param [Hash] **keyword_args update image based on keyword arguments. Keyword arguments are :mime_type,
+        #   :bytes, :file and :url. If :bytes or :file is supplied a :mime_type must also be supplied. If
+        #   :url is supplied the image at the supplied URL is downloaded
+        #
+        #
+        def update(base_64 = nil, **keyword_args)
+          if base_64
+            update_base64(base_64)
+          elsif keyword_args.length.positive?
+            update_from_keywords(keyword_args)
+          else
+            @image_item.update
+          end
+        end
+
+        #
+        # Get the mime type for the image item
+        #
+        # @return [String] mime type for image, e.g. image/png
+        #
+        def mime_type
+          state&.mime_type
+        end
+
+        #
+        # Get the bytes of the image
+        #
+        # @return [Array] Bytes that comprise the image
+        #
+        def bytes
+          state&.get_bytes
+        end
+
+        private
+
+        #
+        # Update from keyword arguments
+        # @param [Hash] **keyword_args update image based on keyword arguments. Keyword arguments are :mime_type,
+        #   :bytes, :file and :url. If :bytes or :file is supplied a :mime_type must also be supplied. If
+        #   :url is supplied the image at the supplied URL is downloaded
+        #
+        #
+        def update_from_keywords(keyword_args)
+          if keyword_args.key?(:mime_type) && keyword_args.key?(:bytes)
+            update_from_parts(mime_type: keyword_args[:mime_type], bytes: keyword_args[:bytes])
+          elsif keyword_args.key?(:mime_type) && keyword_args.key?(:file)
+            update_from_file(mime_type: keyword_args[:mime_type], file: keyword_args[:file])
+          elsif keyword_args.key? :url
+            update_from_url(URI.parse(keyword_args[:url]))
+          else
+            raise ArgumentError, 'Must have both :mime_type and :bytes or :url to update image'
+          end
+        end
+
+        #
+        # Update image from file
+        #
+        # @param [String] mime_type of image
+        # @param [String] file location
+        #
+        #
+        def update_from_file(mime_type:, file:)
+          update_from_parts(mime_type: mime_type, bytes: IO.binread(file))
```

Comment:
Can the mime type be read/derived from the file itself, so that we only need to provide the file name? It would involve utilising some sort of image parsing library.

Reply:
Yes.  https://github.com/minad/mimemagic